horizons/gui/modules/settings.py

```diff
@@ -172,12 +175,15 @@ def apply_settings(self):
 				if entry.callback:
 					entry.callback(old_value, new_value)
 
-		if restart_required:
-			self.show_restart_popup()
-
 		self.hotkey_interface.save_settings()
 		self._settings.apply()
 		self._settings.save()
+
+		if restart_required:
+			if self.restart_promt():
+				horizons.globals.fife.engine.destroy()
+				os.execl(sys.executable, sys.executable, *sys.argv)
```

Comment:
From reading the [documentation](https://docs.python.org/2/library/os.html#os.execl), `os.execv` should be used here.